src/formats/onenote.ts

```diff
@@ -357,17 +408,121 @@ export class OneNoteImporter extends FormatImporter {
-		// so we need to close them using regex
-		const regex = /<object([^>]*)\/>/g;
-		const pageElement = parseHTML(pageHTML.replace(regex, '<object$1></object>'));
+	/**
+	 * Returns a filesystem path for any OneNote entity (e.g. sections or notes)
+	 * Paths are returned in the following format:
+	 * (Export folder)/Notebook/(possible section groups)/Section/(possible pages with a higher level) 
+	 */
+	getEntityPath(entityID: string, currentPath: string, parentEntity?: OnenoteEntityHierarchyModel | undefined): string | null {
+		let returnPath: string | null = null;
+
+		if (!parentEntity) {
+			for (const notebook of this.notebooks) {
+				const path = this.getEntityPath(entityID, `${currentPath}/${notebook.displayName}`, notebook);
+				if (path) {
+					returnPath = path;
+					break;
+				}
+			}
+		}
+		else {
+			if ('sectionGroups' in parentEntity) {
+				const path = this.searchSectionGroups(entityID, currentPath, parentEntity.sectionGroups as SectionGroup[]);
+				if (path !== null) returnPath = path;
+			}
 
+			if ('sections' in parentEntity) {
+				const path = this.searchSections(entityID, currentPath, parentEntity as SectionGroup);
+				if (path !== null) returnPath = path;
+			}
+
+			if ('pages' in parentEntity && parentEntity.pages) {
+				const path = this.searchPages(entityID, currentPath, parentEntity as OnenoteSection);
+				if (path !== null) returnPath = path;
+			}
+		}
+		return returnPath;
+	  }
+
+	private searchPages(entityID: string, currentPath: string, section: OnenoteSection): string | null {
+		let returnPath: string | null = null;
+		// Check if the target page is in the current entity's pages
+		for (let i = 0; i < section.pages!.length; i++) {
+			const page = section.pages![i];
+			const pageContentID = page.contentUrl!.split('page-id=')[1]?.split('}')[0];
+
+			if (page.id === entityID || pageContentID === entityID) {
+				if (page.level === 0) {
+					/* Checks if we have a page leveled below this one. 
+					 * without this line, leveled notes are more scattered:
+					 * ...Section/Example.md, *but* ...Section/Example/Lower level.md
+					 * with this line both files are in one neat directory:
+					 * ...Section/Example/Page.md and ...Section/Example/Lower level.md
+					 */
+					if (section.pages![i+1] && section.pages![i+1].level !== 0) {
+						returnPath = `${currentPath}/${page.title}`;
+					}
+					else returnPath = currentPath;							
+				}
+				else {
+					returnPath = currentPath;
+
+					// Iterate backward to find the parent page
+					for (let i = section.pages!.indexOf(page) - 1; i >= 0; i--) {
+						if (section.pages![i].level === page.level! - 1) {
+							returnPath += '/' + section.pages![i].title;
+							break;
+						}
+					}
+				}
+				break;
+			}
+		}
+		return returnPath;
+	}
+	
+	private searchSectionGroups(entityID: string, currentPath: string, sectionGroups: SectionGroup[]): string | null {
```

Comment:
If you change this method signature to this:

```
private searchSectionGroups(entityID: string, currentPath: string, sectionGroups: SectionGroup[] | OnenoteSection[]): string | null {
```

Then you can remove `searchSections` and replace the places where it was called with this function, and change the last parameter to `parentEntity.sections!`.